packages/grid/src/vaadin-grid.js

```diff
@@ -653,13 +653,28 @@ class Grid extends ElementMixin(
 
     const cell = document.createElement(tagName);
     cell.id = slotName.replace('-content-', '-');
-    cell.setAttribute('tabindex', '-1');
     cell.setAttribute('role', tagName === 'td' ? 'gridcell' : 'columnheader');
 
     const slot = document.createElement('slot');
     slot.setAttribute('name', slotName);
 
-    cell.appendChild(slot);
+    if (isCellButton) {
+      const div = document.createElement('div');
+      div.setAttribute('role', 'button');
+      div.setAttribute('tabindex', '-1');
+      cell.appendChild(div);
+
+      // Patch `focus()` to use the button
+      cell._button = div;
```

Comment:
Would also suggest to make this more specific, `_focusButton` for example.

Reply:
Done.